### PR TITLE
BaSP-TG-56: controllers, model and postman collection update

### PR DESCRIPTION
**Pull request by ArianaMaldonado**
**The following tickets were done in this PR:**

> - TG-51: updating model for tasks
> - TG-56: updating controllers for tasks
> - TG-53: linking Tasks collections (which was ignored because Tasks does not need it)

### CODE REVIEW

File: src/controllers/tasks.js [hunk 1]
```diff
@@ -46,7 +46,11 @@ const getTaskById = async (req, res) => {
```
Comment (lucasrendo) [inline src/controllers/tasks.js:46]: Put method is not updated so neither workedHours or date can be modified (unless it's intentional)

File: src/validations/tasks.js [hunk 2]
```diff
@@ -9,6 +9,11 @@ const validationSchema = joi.object({
     .max(90)
     .required()
     .regex(/^[0-:A-Za-z ",-.]{1,90}$/),
+  workedHours: joi
```
Comment (lucasrendo) [inline src/validations/tasks.js:12]: i can use negative numbers and as many decimals as i want

File: src/validations/tasks.js [hunk 3]
```diff
@@ -9,6 +9,11 @@ const validationSchema = joi.object({
...
+    .number()
+    .required(),
+  date: joi
```
Comment (lucasrendo) [inline src/validations/tasks.js:15]: no need to validate this as it's set by the database with the current time, the client do not send any date to validate

File: src/validations/tasks.js [hunk 4]
```diff
@@ -41,7 +46,11 @@ const validateUpdate = async (req, res, next) => {
       error: true,
     });
   }
-  const existing = await Tasks.findOne({ description: req.body.description.toLowerCase().trim() });
+  const existing = await Tasks.findOne({
+    description: req.body.description.toLowerCase().trim(),
```
Comment (lucasrendo) [inline src/validations/tasks.js:50]: to check for the description only is enough, as if i have 2 tasks with the exact same description it's considered to already exist, no need to also match existing worked hours and date

**Review by lucasrendo — CHANGES_REQUESTED** [review]
Put validation should be updated so that the client can update the worked hours on a task

File: src/controllers/tasks.js [hunk 5]
```diff
@@ -101,10 +101,13 @@ const deleteTask = async (req, res) => {
...
 const updateTask = async (req, res) => {
   try {
-    const targetTask = await Tasks.findById(req.params.id);
+    const targetTask = await Tasks.findByIdAndUpdate(
+      req.params.id,
+      req.body,
+      { new: true },
+    );
     if (targetTask) {
       targetTask.description = req.body.description.toLowerCase();
```
Comment (lucasrendo) [inline src/controllers/tasks.js:110]: is this line necessary?

**Review by lucasrendo — APPROVED** [review]

**Review by EzeRocco — APPROVED** [review]

**Pull request merged**